horovod/common/common.h

```diff
@@ -259,6 +267,7 @@ struct TensorTableEntry {
   // storage complexity of collecting all worker split arrays
   // on coordinator rank.
   std::vector<int32_t> splits;
+  std::shared_ptr<Tensor> received_splits;
```

Comment:
Since `received_splits` is an output, I think it would more general to make `output` a vector with multiple entries rather than add a separate entry. Since this is the only multi-output operation currently, might not be worth the more extensive code changes for this right now, but something to keep in mind.

Reply:
One potential caveat that came to my mind: There are a lot of instances in the code that look like `(void*)e.output->data()`. These would have to be changed to something like `(void*)e.outputs[0]->data()`, but it would then be quite easy to introduce a typo like `(void*)e.outputs->data()` and the compiler would not notice because both std::vector and horovod::common::Tensor have a member function data(). So maybe it would then be worth to change data() and size() in Tensor to less error prone names?

Reply:
That is a good point. If you did want to go down the route of converting `output` to a vector of tensors, did you have better names in mind?

Reply:
I'd rather postpone the naming debate to some future pull request, actually... If we decide to go forward with turning `output` into a vector, should we also turn `input` into a vector and treat `splits` like an input tensor?

As long as there are no other operations with multiple outputs and as long as there are not more than two different "types" of output, I'd personally prefer the code to be a bit simpler all over.

Reply:
Yes, I agree. We can postpone these changes until there are more multi input/output operations.